src/OpenApi/test/Services/OpenApiDocumentServiceTests.Responses.cs

```diff
@@ -0,0 +1,206 @@
+                response =>
+                {
+                    Assert.Equal("201", response.Key);
+                    Assert.Equal("Created", response.Value.Description);
+                },
+                response =>
+                {
+                    Assert.Equal("400", response.Key);
+                    Assert.Equal("Bad Request", response.Value.Description);
+                });
+        });
+    }
+
+    [Fact]
+    public async Task GetOpenApiResponse_SupportsProblemDetailsResponse()
+    {
+        // Arrange
+        var builder = CreateBuilder();
+
+        // Act
+        builder.MapGet("/api/todos", () => { })
+            .WithMetadata(new ProducesResponseTypeMetadata(StatusCodes.Status400BadRequest, typeof(ProblemDetails), ["application/json+problem"]));
+
+        // Assert
+        await VerifyOpenApiDocument(builder, document =>
+        {
+            var operation = Assert.Single(document.Paths["/api/todos"].Operations.Values);
+            var response = Assert.Single(operation.Responses);
+            Assert.Equal("400", response.Key);
+            Assert.Equal("Bad Request", response.Value.Description);
+            Assert.Equal("application/json+problem", response.Value.Content.Keys.Single());
+        });
+    }
+
+    [Fact]
+    public async Task GetOpenApiResponse_SupportsMultipleResponsesForStatusCode()
+    {
+        // Arrange
+        var builder = CreateBuilder();
+
+        // Act
+        builder.MapGet("/api/todos", () => { })
+            // Simulates metadata provided by IEndpointMetadataProvider
+            .WithMetadata(new ProducesResponseTypeMetadata(StatusCodes.Status200OK))
+            // Simulates metadata added via `Produces` call
+            .WithMetadata(new ProducesResponseTypeMetadata(StatusCodes.Status200OK, typeof(string), ["text/plain"]));
+
+        // Assert
+        await VerifyOpenApiDocument(builder, document => {
+            var operation = Assert.Single(document.Paths["/api/todos"].Operations.Values);
+            var response = Assert.Single(operation.Responses);
+            Assert.Equal("200", response.Key);
+            Assert.Equal("OK", response.Value.Description);
+            var content = Assert.Single(response.Value.Content);
+            Assert.Equal("text/plain", content.Key);
+        });
+    }
+
+    [Fact]
+    public async Task GetOpenApiResponse_SupportsMultipleResponseTypesWithTypeForStatusCode()
+    {
+        // Arrange
+        var builder = CreateBuilder();
+
+        // Act
+        builder.MapGet("/api/todos", () => { })
+            // Simulates metadata provided by IEndpointMetadataProvider
+            .WithMetadata(new ProducesResponseTypeMetadata(StatusCodes.Status200OK, typeof(Todo), ["application/json"]))
+            // Simulates metadata added via `Produces` call
+            .WithMetadata(new ProducesResponseTypeMetadata(StatusCodes.Status200OK, typeof(TodoWithDueDate), ["application/json"]));
```

Comment:
Is the `typeof(someType)` not important from OpenApi perspective?
Should we have a test for 2 different content type metadatas for the same status code?

Reply:
> Is the typeof(someType) not important from OpenApi perspective?

The response type is important for schema generation, which isn't included in the implementation yet. Once I add that in a separate PR, I'll include test cases for these things.

> Should we have a test for 2 different content type metadatas for the same status code?

`GetOpenApiResponse_SupportsMultipleResponseTypesWitDifferentContentTypes` cover this unless you are talking about something else? Do you mean something like:

```

        builder.MapGet("/api/todos", () => { })
            .WithMetadata(new ProducesResponseTypeMetadata(StatusCodes.Status200OK, typeof(Todo), ["application/json"]))
			.WithMetadata(new ProducesResponseTypeMetadata(StatusCodes.Status200OK, typeof(TodoWithDueDate), ["application/xml"]));
 ```

Reply:
> Do you mean something like:

Yeah, that sample

Reply:
Yep, in that case the only metadata that will be used in the last one based on the LIFO rules for resolving metadata so you'd end up with `application/json` and the `TodoWithDueDate` schema. I added a test for this in [`6447c73` (#55020)](https://github.com/dotnet/aspnetcore/pull/55020/commits/6447c736cf7a15da8fbb0fdddc53cf57bcca2637)